aquadoggo/src/replication/strategies/naive.rs

```diff
@@ -48,7 +58,10 @@ impl NaiveStrategy {
         let mut messages = Vec::new();
 
         let local_log_heights = self.local_log_heights(store).await;
-        let remote_needs = diff_log_heights(&local_log_heights, remote_log_heights);
+        let remote_needs = diff_log_heights(
+            &local_log_heights.to_owned().into_iter().collect(),
```

Comment:
This was the cause of https://github.com/p2panda/aquadoggo/issues/392 .... :sob: